Modules/_testcapi/list.c

```diff
@@ -0,0 +1,97 @@
+#include "parts.h"
+#include "util.h"
+
+static PyObject *
+list_check(PyObject* Py_UNUSED(module), PyObject *obj)
+{
+    NULLABLE(obj);
+    return PyLong_FromLong(PyList_Check(obj));
+}
+
+static PyObject *
+list_check_exact(PyObject* Py_UNUSED(module), PyObject *obj)
+{
+    NULLABLE(obj);
+    return PyLong_FromLong(PyList_CheckExact(obj));
+}
+
+static PyObject *
+list_new(PyObject* Py_UNUSED(module), PyObject *obj)
+{
+    return PyList_New(PyLong_AsSsize_t(obj));
+}
+
+static PyObject *
+list_size(PyObject *Py_UNUSED(module), PyObject *obj)
+{
+    NULLABLE(obj);
+    RETURN_SIZE(PyList_Size(obj));
+}
+
+static PyObject *
+list_getitem(PyObject *Py_UNUSED(module), PyObject *args)
+{
+    PyObject *obj;
+    Py_ssize_t i;
+    if (!PyArg_ParseTuple(args, "On", &obj, &i)){
+        return NULL;
+    }
+    NULLABLE(obj);
+    return PyList_GetItem(obj, i);
```

Comment:
`PyList_GetItem()` returns a borrowed reference. Call `Py_XNewRef()` for the result.

Reply:
I have seen examples of how to use `Py_XNewRef` and I am a bit confused. Existing examples use it on the value like this 
```
Py_INCREF(key);
    node->key = key;

    Py_INCREF(data);
    node->data = data;
```
For any object that we are returning/assigning we need to use `Py_XNewRef` so that it will increase the ref count and then return a new object. Otherwise, we are just returning the original reference. 

Please correct me if my understanding is correct.